src/engine/source/cmds/src/start.cpp

```diff
@@ -0,0 +1,294 @@
+
+        catalog = std::make_shared<api::catalog::Catalog>(catalogConfig);
+        WAZUH_LOG_INFO("Catalog initialized.");
+
+        api::catalog::cmds::registerAllCmds(catalog, server->getRegistry());
+        WAZUH_LOG_DEBUG("Catalog API registered.")
+
+        envManager = std::make_shared<router::EnvironmentManager>(builder, server->getEventQueue(), threads);
+        g_exitHanlder.add([envManager]() { envManager->delAllEnvironments(); });
+        WAZUH_LOG_INFO("Environment manager initialized.");
+
+        // Register the API command
+        server->getRegistry()->registerCommand("env", envManager->apiCallback());
+        WAZUH_LOG_DEBUG("Environment manager API registered.")
+
+        // Register Configuration API commands
+        api::config::cmds::registerCommands(server->getRegistry(), confManager);
+        WAZUH_LOG_DEBUG("Configuration manager API registered.");
+
+        // Up default environment
+        auto error = envManager->addEnvironment(environment);
+        if (!error)
+        {
+            envManager->startEnvironment(environment);
+        }
+        else
+        {
+            WAZUH_LOG_WARN("An error occurred while creating the default environment \"{}\": {}.",
+                           environment,
+                           error.value().message);
+            WAZUH_LOG_WARN("Engine running without active environment.")
+        }
+    }
+    catch (const std::exception& e)
+    {
+        WAZUH_LOG_ERROR("While initializing modules: ", utils::getExceptionStack(e));
+        g_exitHanlder.execute();
+        return;
+    }
+
+    // Start server
+    try
+    {
+        server->run();
+    }
+    catch (const std::exception& e)
+    {
+        WAZUH_LOG_ERROR("While server running: {}.", utils::getExceptionStack(e));
+        g_exitHanlder.execute();
+        return;
+    }
+    g_exitHanlder.execute();
+}
+
+void configure(CLI::App_p app)
+{
+    auto serverApp = app->add_subcommand("server", "Start/Stop a Wazuh engine instance.");
+    serverApp->require_subcommand(1);
+    auto options = std::make_shared<Options>();
+
+    // Log level
+    serverApp
+        ->add_option(
+            "--log_level", options->logLevel, "Sets the logging level: 0 = Debug, 1 = Info, 2 = Warning, 3 = Error")
+        ->default_val(ENGINE_LOG_LEVEL)
+        ->check(CLI::Range(0, 3))
+        ->envname(ENGINE_LOG_LEVEL_ENV);
+    // Log output
+    serverApp->add_option("--log_output", options->logOutput, "Sets the logging output")
+        ->default_val(ENGINE_LOG_OUTPUT)
+        ->check(CLI::ExistingFile)
+        ->envname(ENGINE_LOG_OUTPUT_ENV);
+
+    // Server
+    // Endpoints
+    serverApp->add_option("--event_endpoint", options->eventEndpoint, "Sets the events server socket address.")
+        ->default_val(ENGINE_EVENT_SOCK)
+        ->envname(ENGINE_EVENT_SOCK_ENV);
+    serverApp->add_option("--api_endpoint", options->apiEndpoint, "Sets the API server socket address.")
```

Comment:
we should unify all options, to `api_socket` or `api_endpoint` including help messages

Reply:
changed to socket